c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1252,38 +1252,43 @@ class DeviceCachingAllocator {
       });
     }
 
-    block->allocated = true;
-    block->requested_size = orig_size;
+    // Skip unmapped expandable segments when reapplying checkpoints
```

Comment:
This function is called `allocate_found_block` so it would be very surprising that in certain cases it doesn't actually make the block passed to it allocated. I think it would be better to factor out the part that is specifically needed for restoring graph state and giving a name based what the part actually does rather than having a function that doesn't do what its named to do.

Reply:
With expandable segments we have mapped and unmapped blocks when reapplying the checkpoint state so it needs to replay all the logic for setting block state and updating memory stats in `alloc_found_block` for the mapped blocks (and most of the same logic for the unmapped blocks). I think factoring out to something like `restore_block_state()` would look like duplicating `alloc_found_block()` as it is right now.

Happy to do that but I wouldn't expect anyone to call `alloc_found_block` with an unmapped block, but if they did this would properly increment reserved memory while leaving `block.allocated=False` and active memory unchanged.

Reply:
Looking at this more -- the entire alloc_found_block function is a no-op for unmapped blocks. In particular, there can never be a case where there are two unmapped blocks in a row, so an unmapped block will never be splittable, and then the rest of the logic is guarded away. I think a better approach here would be to just skip unmapped blocks when restoring checkpoints. The code to restore allocation state doesn't correctly edit mapped/unmapped state anyway.